Hazel/src/Hazel/Scene/SceneSerializer.cpp

```diff
@@ -82,70 +82,86 @@ namespace Hazel {
+		out << YAML::Key << "OrthographicFar" << YAML::Value << camera.GetOrthographicFarClip();
+		out << YAML::EndMap; // Camera
+
+		out << YAML::Key << "Primary" << YAML::Value << cameraComponent.Primary;
+		out << YAML::Key << "FixedAspectRatio" << YAML::Value << cameraComponent.FixedAspectRatio;
+
+		out << YAML::EndMap; // CameraComponent
+	}
 
-		if (entity.HasComponent<CameraComponent>())
-		{
-			out << YAML::Key << "CameraComponent";
-			out << YAML::BeginMap; // CameraComponent
-
-			auto& cameraComponent = entity.GetComponent<CameraComponent>();
-			auto& camera = cameraComponent.Camera;
-
-			out << YAML::Key << "Camera" << YAML::Value;
-			out << YAML::BeginMap; // Camera
-			out << YAML::Key << "ProjectionType" << YAML::Value << (int)camera.GetProjectionType();
-			out << YAML::Key << "PerspectiveFOV" << YAML::Value << camera.GetPerspectiveVerticalFOV();
-			out << YAML::Key << "PerspectiveNear" << YAML::Value << camera.GetPerspectiveNearClip();
-			out << YAML::Key << "PerspectiveFar" << YAML::Value << camera.GetPerspectiveFarClip();
-			out << YAML::Key << "OrthographicSize" << YAML::Value << camera.GetOrthographicSize();
-			out << YAML::Key << "OrthographicNear" << YAML::Value << camera.GetOrthographicNearClip();
-			out << YAML::Key << "OrthographicFar" << YAML::Value << camera.GetOrthographicFarClip();
-			out << YAML::EndMap; // Camera
-
-			out << YAML::Key << "Primary" << YAML::Value << cameraComponent.Primary;
-			out << YAML::Key << "FixedAspectRatio" << YAML::Value << cameraComponent.FixedAspectRatio;
-
-			out << YAML::EndMap; // CameraComponent
-		}
+	template<>
+	static void SerializeComponent(
+		YAML::Emitter& out, const SpriteRendererComponent& spriteRendererComponent)
+	{
+		out << YAML::Key << "SpriteRendererComponent";
+		out << YAML::BeginMap; // SpriteRendererComponent
 
-		if (entity.HasComponent<SpriteRendererComponent>())
-		{
-			out << YAML::Key << "SpriteRendererComponent";
-			out << YAML::BeginMap; // SpriteRendererComponent
+		out << YAML::Key << "Color" << YAML::Value << spriteRendererComponent.Color;
 
-			auto& spriteRendererComponent = entity.GetComponent<SpriteRendererComponent>();
-			out << YAML::Key << "Color" << YAML::Value << spriteRendererComponent.Color;
+		out << YAML::EndMap; // SpriteRendererComponent
+	}
 
-			out << YAML::EndMap; // SpriteRendererComponent
-		}
+	template<typename T>
+	static void TrySerializeComponent(YAML::Emitter& out, const Entity& entity)
+	{
+		if (entity.HasComponent<T>())
+			SerializeComponent(out, entity.GetComponent<T>());
+	}
+
+	static void SerializeEntity(YAML::Emitter& out, Entity entity)
+	{
+		out << YAML::BeginMap; // Entity
+		out << YAML::Key << "Entity" << YAML::Value << "12837192831273"; // TODO: Entity ID goes here
+
+		TrySerializeComponent<TagComponent>(out, entity);
+		TrySerializeComponent<TransformComponent>(out, entity);
+		TrySerializeComponent<CameraComponent>(out, entity);
+		TrySerializeComponent<SpriteRendererComponent>(out, entity);
```

Comment:
More a follow-up on a previous comment, if the previous suggestion is merged, then these require renaming
```suggestion
		SerializeComponent<TagComponent>(out, entity);
		SerializeComponent<TransformComponent>(out, entity);
		SerializeComponent<CameraComponent>(out, entity);
		SerializeComponent<SpriteRendererComponent>(out, entity);
```